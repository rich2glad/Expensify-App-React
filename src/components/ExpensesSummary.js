import React from 'react'
import numeral from 'numeral'
import {connect} from 'react-redux';
import selectExpenses from '../selectors/expenses'
import selectExpensesTotal from '../selectors/expenses-total'

export const ExpensesSummary =({expenseCount,expenseTotal})=>{
    const expenseWord = expenseCount ===1?'expense':'expenses';
    const formattedTotalExpense = numeral(expenseTotal/100).format('$0,00.00');
    return(
    <div>
    <h1> viewing {expenseCount} {expenseWord} totalling {formattedTotalExpense}</h1>        
    </div>
)
}
const mapStateToProps=(state)=>{
    const visibleExpenses = selectExpenses(state.expenses,state.filters)
    return {
        expenseCount: visibleExpenses.length,
        expenseTotal: selectExpensesTotal(visibleExpenses)
    };
}

export default connect(mapStateToProps)(ExpensesSummary)