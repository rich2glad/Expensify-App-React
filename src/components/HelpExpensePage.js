import React from 'react';
const HelpExpensePage=()=>(
    <div>
    This is Help Expense Component</div>
);
export default HelpExpensePage