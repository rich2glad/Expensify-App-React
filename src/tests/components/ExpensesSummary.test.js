import React from 'react';
import {shallow} from 'enzyme'
import {ExpensesSummary} from '../../components/ExpensesSummary'

test('should correctly render Expense with 1 expense',()=>{
    const wrapper = shallow(<ExpensesSummary  expenseCount={1} expensesTotal={234.45} />);
    expect(wrapper).toMatchSnapshot();
})

test('should correctly render Expense with multiple expense',()=>{
    const wrapper = shallow(<ExpensesSummary  expenseCount={23} expensesTotal={23445.90} />);
    expect(wrapper).toMatchSnapshot();
})