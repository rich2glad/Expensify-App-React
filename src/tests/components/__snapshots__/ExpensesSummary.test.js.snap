// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render Expense with 1 expense 1`] = `
<div>
  <h1>
     viewing 
    1
     
    expense
     totalling 
    $0.00
  </h1>
</div>
`;

exports[`should correctly render Expense with multiple expense 1`] = `
<div>
  <h1>
     viewing 
    23
     
    expenses
     totalling 
    $0.00
  </h1>
</div>
`;
